src/core/symbology-ng/qgssymbollayerutils.cpp

```diff
@@ -2514,7 +2515,15 @@ bool QgsSymbolLayerUtils::functionFromSldElement( QDomElement &element, QString
     return false;
   }
 
+  // check if it is a single string value
+  if ( elem.hasChildNodes() &&
+       elem.firstChild().nodeType() == QDomNode::TextNode )
+  {
+    function = elem.firstChild().nodeValue();
```

Comment:
Doesn't this need to be sent through `QgsOgcUtils::expressionFromOgcFilter( elem );`?